src/utils/helpers/DetailsDrawHelpers.ts

```diff
@@ -179,13 +221,31 @@ function createDiagrams(
     const consumersDiagram = createDiagram(consumersTree, width, height, rootNodeYOffset, true);
     const providersDiagram = createDiagram(providersTree, width, height, rootNodeYOffset);
 
-    if (providersDiagram) {
-        container.append<SVGSVGElement>(() => providersDiagram);
-    }
+    createRootNode(container, width, height, rootNodeName, Boolean(providersDiagram), Boolean(consumersDiagram));
 
-    if (consumersDiagram) {
-        container.append<SVGSVGElement>(() => consumersDiagram);
-    }
+    setTimeout(() => {
```

Comment:
why this is async?

Reply:
changed it from setTimeout to use promises and async/await

It is needed though because I cannot get BBox of an element right after it is created - I have to wait before it gets rendered in DOM.